tools/missing-test-detector/changed.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	newProvider "google/provider/new/google-beta"
+	oldProvider "google/provider/old/google-beta"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// A map of field names with whether the field is covered as values.
+type FieldCoverage map[string]any
+
+// Returns a map with resource names as keys and field coverage maps as values.
+func changedResourceFields() map[string]FieldCoverage {
+	oldResourceMap := oldProvider.ResourceMap()
+	newResourceMap := newProvider.ResourceMap()
+
+	return resourceMapChanges(oldResourceMap, newResourceMap)
+}
+
+func resourceMapChanges(oldResourceMap, newResourceMap map[string]*schema.Resource) map[string]FieldCoverage {
+	changes := make(map[string]FieldCoverage)
+	for resourceName, newResource := range newResourceMap {
+		if fields := changedFields(oldResourceMap[resourceName], newResource, false); len(fields) > 0 {
+			changes[resourceName] = fields
+		}
+
+	}
+	return changes
+}
+
+func changedFields(oldResource, newResource *schema.Resource, nested bool) FieldCoverage {
+	fields := make(FieldCoverage)
+	for fieldName, newFieldSchema := range newResource.Schema {
+		if fieldName == "project" && !nested {
```

Comment:
could you add a comment here explaining why this case is skipped?

Reply:
Done.